internal/scanner/scanner.go

```diff
@@ -57,26 +69,91 @@ func (s *Scanner) ScanNamespace(namespace string) error {
 
 	// TODO this is for debugging, it should be remove in future steps!
 	for _, resource := range auditableResources {
-		fmt.Println("Policies: ")
+		log.Debug().Msg("Policies: ")
 		for _, policy := range resource.Policies {
-			fmt.Println(policy.GetName())
+			url, err := s.resourcesFetcher.GetPolicyServerURLRunningPolicy(context.Background(), policy)
+			if err != nil {
+				log.Debug().Err(err).Msg("CANNOT GET POLICY SERVER URL")
+				continue
+			}
+			log.Debug().Msgf("POLICY SERVER URL: %s", url.String())
 		}
 		for _, resource := range resource.Resources {
-			fmt.Println(resource)
-			fmt.Println("........")
+			log.Debug().Msgf("Opa!")
+			log.Debug().Dict("resource", zerolog.Dict().
+				Str("name", resource.GetName()).
+				Str("kind", resource.GetKind()).
+				Str("namespace", resource.GetNamespace()).
+				Str("UID", string(resource.GetUID()))).
+				Msg("auditable resource")
 		}
-		fmt.Println("---------------------")
 	}
 
-	// TODO for next steps:
 	// Iterate through all auditableResources. Each item contains a list of resources and the policies that would need
 	// to evaluate them. You need to create the AdmissionReview request using the resource which is an unstructured.
 	// unstructured nested functions might help with that https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured
 	// or the UnstructuredContent() method that returns a map
+	for i := range auditableResources {
+		auditResource(&auditableResources[i], &s.resourcesFetcher, &s.httpClient)
+	}
 
 	return nil
 }
 
+func auditResource(resource *resources.AuditableResources, resourcesFetcher *ResourcesFetcher, httpClient *http.Client) {
+	for _, policy := range resource.Policies {
+		url, err := (*resourcesFetcher).GetPolicyServerURLRunningPolicy(context.Background(), policy)
+		if err != nil {
+			// TODO what's the better thing to do here?
+			log.Error().Err(err)
+			continue
```

Comment:
I can't think of anything better to be fair.

Reply:
To be honest, I'm trying nothing. Just logging the error for now. As we are not populating the report with the errors yet I considered that is fine just logging the error here for now. Because sending the error upwards will not make that difference once we probably will log the error there as well. Thus, I decided to leave the logging here and move it (if necessary) after. That's why I leave the TODO flag there.

Reply:
Yep, sounds good! Resolving.